daemon/openrazer_daemon/hardware/keyboards.py

```diff
@@ -344,6 +345,52 @@ class RazerBlackWidowUltimate2013(_MacroKeyboard):
     DEVICE_IMAGE = "https://assets.razerzone.com/eeimages/support/products/245/438_blackwidow_ultimate_2014.png"
 
 
+class RazerBlackWidowv3ProUSB(_MacroKeyboard):
+    """
+    Class for the Razer BlackWidow v3 PRO usb
+    """
+    EVENT_FILE_REGEX = re.compile(r'.*Razer_Razer_BlackWidow_V3_Pro(-if01)?-event-kbd')
+
+    USB_VID = 0x1532
+    USB_PID = 0x025A
+    DEDICATED_MACRO_KEYS = True
+    MATRIX_DIMS = [6, 22]
+    METHODS = ['get_device_type_keyboard', 'get_game_mode', 'set_game_mode', 'set_macro_mode', 'get_macro_mode',
+               'get_macro_effect', 'set_macro_effect', 'bw_set_pulsate', 'bw_set_static', 'get_macros', 'delete_macro', 'add_macro']
+
+    DEVICE_IMAGE = "https://dl.razerzone.com/src/3809-1-EN-v1.png"
+
+
+class RazerBlackWidowv3ProBluetooth(RazerBlackWidowv3ProUSB):
+    """
+    Class for the Razer BlackWidow v3 PRO bluetooth
+    """
+    USB_PID = 0x025B
+    METHODS = RazerBlackWidowv3ProUSB.METHODS + ['set_charge_effect', 'set_charge_colour']
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._battery_manager = _BatteryManager(self, self._device_number, 'Razer BlackWidow v3 PRO bluetooth')
```

Comment:
The BatteryManager needs the `getBattery` function to work (as seen in the CI failure), so you probably want to add these methods to the METHODS array:
```
               'get_battery', 'is_charging', 'set_idle_time', 'set_low_battery_threshold',
```
You need to add the needed kernel parts as well and then validate that the methods are working

Reply:
> The BatteryManager needs the `getBattery` function to work (as seen in the CI failure), so you probably want to add these methods to the METHODS array:
> 
> ```
>                'get_battery', 'is_charging', 'set_idle_time', 'set_low_battery_threshold',
> ```
> 
> You need to add the needed kernel parts as well and then validate that the methods are working

@z3ntu 
I have problems with battery methods, can you help me?
```Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 954, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3/dist-packages/openrazer_daemon/misc/battery_notifier.py", line 105, in run
    self.notify_battery()
  File "/usr/lib/python3/dist-packages/openrazer_daemon/misc/battery_notifier.py", line 75, in notify_battery
    battery_level = self._get_battery_func()
  File "/usr/lib/python3/dist-packages/openrazer_daemon/dbus_services/dbus_methods/mamba.py", line 15, in get_battery
    with open(driver_path, 'r') as driver_file:
FileNotFoundError: [Errno 2] No such file or directory: '/sys/devices/pci0000:00/0000:00:14.0/usb1/1-4/1-4:1.2/0003:1532:025A.0030/charge_level'
```

Reply:
According to the driver this should exist. Are you sure you have the latest version loaded?